src/commands/publish/route.rs

```diff
@@ -28,103 +34,71 @@ impl Route {
-            res.text()?
-        );
-        failure::bail!(msg)
-    }
+    let client = api_client(user, HttpApiClientConfig::default())?;
 
-    let routes_response: RoutesResponse = serde_json::from_str(&res.text()?)?;
+    let routes: Vec<Route> = match client.request(&ListRoutes {
+        zone_identifier: &target.zone_id.as_ref().unwrap(),
+    }) {
+        Ok(success) => success.result.iter().map(|r| Route::from(r)).collect(),
+        Err(e) => failure::bail!("{}", format_error(e, None)), // TODO: add suggestion fn
+    };
 
-    Ok(routes_response.result)
+    Ok(routes)
 }
 
 fn create(user: &GlobalUser, target: &Target, route: &Route) -> Result<(), failure::Error> {
-    let client = http::auth_client(None, user);
-    let body = serde_json::to_string(&route)?;
-
-    let routes_addr = get_routes_addr(target)?;
-
-    info!("Creating your route {:#?}", &route.pattern,);
-    let mut res = client
-        .post(&routes_addr)
-        .header(CONTENT_TYPE, "application/json")
-        .body(body)
-        .send()?;
-
-    if !res.status().is_success() {
-        let msg;
-        if res.status().as_u16() == 10020 {
-            msg = format!(
-            "{} A worker with a different name was previously deployed to `{}`. If you would like to overwrite that worker, you will need to change `name` in your `wrangler.toml` to match the currently deployed worker, or navigate to https://dash.cloudflare.com/workers and rename or delete that worker.\n",
-            emoji::WARN,
-            serde_json::to_string(&route)?
-            );
-        } else {
-            msg = format!(
-                "{} There was an error creating your route.\n Status Code: {}\n Msg: {}",
-                emoji::WARN,
-                res.status(),
-                res.text()?
-            );
-        }
-
-        failure::bail!(msg)
+    let client = api_client(user, HttpApiClientConfig::default())?;
+
+    log::info!("Creating your route {:#?}", &route.pattern,);
+    match client.request(&CreateRoute {
+        zone_identifier: &target.zone_id.as_ref().unwrap(),
+        params: CreateRouteParams {
+            pattern: route.pattern.clone(),
+            script: route.script.clone(),
+        },
+    }) {
+        Ok(_success) => Ok(()),
+        Err(e) => failure::bail!("{}", format_error(e, Some(&routes_error_help))),
     }
-    Ok(())
 }
 
-fn get_routes_addr(target: &Target) -> Result<String, failure::Error> {
-    if let Some(zone_id) = &target.zone_id {
-        return Ok(format!(
-            "https://api.cloudflare.com/client/v4/zones/{}/workers/routes",
-            zone_id
-        ));
+fn routes_error_help(error_code: u16) -> &'static str {
+    match error_code {
+        10020 => r#"
+            A worker with a different name was previously deployed to the specified route.
+            If you would like to overwrite that worker,
+            you will need to change `name` in your `wrangler.toml` to match the currently deployed worker,
```

Comment:
Also out of scope for this - I wonder if it's possible to return the name of the worker that's currently associated with that route.

Reply:
i actually have a ticket open to the API with that request, because as it stands today, no it is not possible without making another get request. I will likely be optimizing these calls to make a single get request and keep that set around, in which case we can totally do something like this, but i am also thinking of trying to modify the api to return the appropriate error message.